File: gr/deep-trip.hs
-- deep trip (jmcc) #9

import Sound.SC3 {- hsc3 -}

import qualified Sound.SC3.UGen.External.RDU as RDU  {- sc3-rdu -}
import qualified Sound.SC3.Lang.Control.OverlapTexture as O {- hsc3-lang -}

deep_trip :: UGen
deep_trip =
    let f = midiCPS (lfNoise1 'α' KR (rand 'β' 0 0.3) * 60 + 70)
        a'' = sinOsc KR (rand 'γ' 0 40) 0 * 0.1
        a' = max 0 (lfNoise1 'δ' KR (rand 'ε' 0 8) * a'')
        a = lfNoise2 'ζ' AR (f * rand 'η' 0 0.5) * a'
        z = sinOsc AR f 0 * a
        s = pan2 z (lfNoise1 'θ' KR (rand 'ι' 0 5)) 1
        c0 = combN s 0.5 (RDU.randN 2 'κ' 0.3 0.5) 20
        c1 = combN s 0.5 (RDU.randN 2 'λ' 0.3 0.5) 20
        in s + c0 + c1

main :: IO ()
main = O.overlapTextureU (4,12,4,maxBound) deep_trip

File: gr/deep-trip.fs
\ deep trip (jmcc) #9

: f 0 0.3 Rand.ir LFNoise1.kr 60 * 70 + MIDICPS ;
: s f 0 SinOsc.ar f 0 0.5 Rand.ir * LFNoise2.ar 0 0 8 Rand.ir LFNoise1.kr
    0 40 Rand.ir 0 SinOsc.kr 0.1 * * Max * * 0 5 Rand.ir LFNoise1.kr 1 Pan2 ;
: c s 0.5 0.3 0.5 Rand.ir 2 clone 20 CombN ;
s c + c + 4 12 4 inf overlap-texture
